latex/epoch.tex

```diff
@@ -484,58 +473,74 @@ \subsection{Rewards Distribution}
-      & \text{reward one pool} \\
-      & \fun{rewardOnePool}~\var{pp}~\var{R}~\var{n}~\var{poolHK}~\var{pool}
-        ~\var{actgr}~\var{avgs}~\var{tot} = \\
-      & ~~~(\{ (\AddrRWD~hk)\mapsto ( \fun{indivRew}~\var{poolR}~\var{pool}
-                                 ~(\sigma\div\var{tot})~(\var{c}\div\var{tot})~(hk ==\var{poolHK})) \\
-      & ~~~~~~~\mid hk\mapsto c\in\var{actgr}\},~\var{avg})\\
+      & \fun{rewardOnePool} \in \PParams \to \Coin \to \N \to \HashKey \to \PoolParam\\
+      & ~~~\to \Stake \to \Avgs \to \Coin \to \StakeKeys \to (\AddrRWD \mapsto \Coin)\times\Coin \\
+      & \rewardOnePool{pp}{R}{n}{poolHK}{pool}{stake}{avgs}{tot}{stkeys} =
+          (\var{rewards},~\var{unrealized})\\
       & ~~~\where \\
-      & ~~~~~~~\var{pstake} = \sum_{\_\mapsto t\in\var{actgr}} t \\
-      & ~~~~~~~\sigma = \var{pstake} \div tot \\
+      & ~~~~~~~\var{pstake} = \sum_{\_\mapsto t\in\var{stake}} t \\
+      & ~~~~~~~\sigma = \var{pstake} / tot \\
       & ~~~~~~~\var{\overline{N}} = \sigma * \slotsPer\\
-      & ~~~~~~~(\_, \_, p) = \fun{poolCosts}~pool \\
-      & ~~~~~~~p_{r} = p \div tot \\
+      & ~~~~~~~(\_, \_, \var{pledge}) = \fun{poolCosts}~pool \\
+      & ~~~~~~~p_{r} = \var{pledge} / \var{tot} \\
       & ~~~~~~~maxP =
       \begin{cases}
-          \fun{maxPool}~\var{pp}~\var{R}~\sigma~\var{p_r}&
-              p \leq \var{pstake}\\
-          0 & \text{otherwise.}
-        \end{cases} \\
-      & ~~~~~~~(\var{poolR},~\var{avg}) = \fun{poolRew}~\var{pp}~\var{hk}~\var{n}~\var{\overline{N}}~\var{avgs}~\var{maxP} \\
-      \nextdef
-      & \fun{reward} \in \BlocksMade \to \PParams \to \Coin \\
-      & ~~ \to \DPState \to \UTxO \to (\AddrRWD \mapsto \Coin)\times\Distr
-      & \text{staking rewards} \\
-      & \fun{reward}~\var{prod}~\var{pp}~\var{R}~\var{dpstate}~\var{utxo} = \\
-      & ~~~\left(
-           \bigcup_{hk\mapsto(rew,~\_)\in\var{results}} hk\mapsto\var{rew},
-           ~~\bigcup_{hk\mapsto(\_,~\var{avg})\in\var{results}} hk\mapsto\var{avg}
-           \right)\\
+        \fun{maxPool}~\var{pp}~\var{R}~\sigma~\var{p_r}&
+        \var{pledge} \leq \var{pstake}\\
+        0 & \text{otherwise.}
+      \end{cases} \\
+      & ~~~~~~~\var{poolR} = \poolReward{pp}{hk}{n}{\overline{N}}{avgs}{maxP} \\
+      & ~~~~~~~\var{mRewards} = \left\{
+                                  \addrRw~hk\mapsto\mReward{poolR}{pool}{\frac{c}{tot}}{\sigma}
+                                  ~\Big\vert~
+                                  hk\mapsto c\in\var{stake},~~hk \neq\var{poolHK}
+                               \right\}\\
+      & ~~~~~~~\var{lReward} = \lReward{poolR}{pool}{\frac{\var{stake}~\var{poolHK}}{tot}}{\sigma} \\
+      & ~~~~~~~\var{poolAcnt} = \fun{poolAcntHK}~\var{pool} \\
+      & ~~~~~~~\var{rewards} =
+                 \begin{cases}
+                   \var{mReward}\cup\{\addrRw{\var{poolAcnt}}\mapsto\var{lReward}\}
```

Comment:
should this be `mRewards` ?

Reply:
no, this is the rewards for the pool operator and pool owners.  In Berlin we decided that all the rewards associated with these stake keys go into a single reward address specified in the pool registration certificate, assuming it is registered.  If it is not registered, it goes to the treasury.